staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -587,6 +587,55 @@ func TestTimeBucketWatchersBasic(t *testing.T) {
 	}
 }
 
+func TestCacherNoLeakWhenNoUpdate(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchBookmark, true)()
+	backingStorage := &dummyStorage{}
+	cacher, _, err := newTestCacher(backingStorage, 1000)
+	if err != nil {
+		t.Fatalf("Couldn't create cacher: %v", err)
+	}
+	defer cacher.Stop()
+
+	// Wait until cacher is initialized.
+	cacher.ready.wait()
+	pred := storage.Everything
+	pred.AllowWatchBookmarks = true
+
+	ctx, _ := context.WithTimeout(context.Background(), 3*time.Second)
+	w, err := cacher.Watch(ctx, "pods/ns", "0", pred)
+	if err != nil {
+		t.Fatalf("Failed to create watch: %v", err)
+	}
+	timeoutCh := time.After(2 * time.Second)
+	lastObservedRV := uint64(0)
+	for {
+		select {
+		case event, ok := <-w.ResultChan():
+			if !ok {
+				t.Fatal("Unexpected closed")
+			}
+			rv, err := cacher.versioner.ObjectResourceVersion(event.Object)
+			if err != nil {
+				t.Errorf("failed to parse resource version from %#v", event.Object)
+			}
+			if event.Type == watch.Bookmark {
```

Comment:
I don't think that's the right thing to do - this test should Errorf() if any event was emitted (this is a whitebox test).